### PR TITLE
Webaudio player improvements

### PR DESCRIPTION
**Pull request by joggienl**
- Added GainNode to be able to control player volume.
- Some small fixes for pause/resume
- Some helper functions for advanced usage
- Changelog update

### CODE REVIEW

File: src/WebAudio/Player.js [hunk 1]
```diff
@@ -128,7 +157,9 @@ export default class Player extends AudioContext {
   play() {
     const offset = this.pausedAt;
 
-    this.createBufferSource();
+    if (offset !== 0) {
```
Comment (kvdb) [inline src/WebAudio/Player.js]: I guess this line deserves a comment to why this is code is here

**Comment by kvdb** [comment]
rest is fine

File: src/WebAudio/Player.js (hunk 1 above)
Comment (joggienl) [reply src/WebAudio/Player.js]: only create a bufferSource if the the offset is not 0, or do you want to know something else here?

Comment (kvdb) [reply src/WebAudio/Player.js]: Let's take a step back. I'm not referring to just this specific line. I think I'm looking for a more high level description on when AudioBufferSourceNode or when AudioContext is used. What's the purpose of them, when should we be using one over the other. Knowing this, I can probably understand the code in line 160...162 without further detailed comments.

Comment (joggienl) [reply src/WebAudio/Player.js]: I see what you mean. I will create an issue (in YT) do do this later. Documentation on the player can be improved, but the Audio*Node/Context stuff is in general, just general Web Audio API stuff, so nothing special tbh.

**Pull request merged**